moviecenter/main.py

```diff
@@ -1,18 +1,27 @@
 from fastapi import FastAPI, HTTPException
 
-from .movies import Movie, add_movie, delete_movie, get_all_movies, get_movie_by_name
+from .movies import (
+    Movie,
+    MovieMongoClient,
+    add_movie,
+    delete_movie,
+    get_all_movies,
+    get_movie_by_name,
+)
 
 app = FastAPI()
 
+db_client = MovieMongoClient("mongodb://mongodb:27017")
```

Comment:
I strongly advise initiating connections on module level.
This should be done in a main/setup/create_app function and injected into the handlers using, for example, fastify's annotation-based dependency injection.

Connecting on the module level severly harms your flexibility and makes it to easy for every piece of code in your app to just snatch it and access the DB without you noticing.

That's an especially nasty case of using global (although immutable) variables instead of argument passing

Reply:
You are right, I'll add this in a different PR